libvoikko/typescript/libvoikko.d.ts

```diff
@@ -0,0 +1,198 @@
+export = Libvoikko;
+
+declare class Libvoikko {
+  constructor();
+  init(language: string): Voikko;
+}
+
+declare interface GrammarError {
+  errorCode: number;
+  startPos: number;
+  errorLen: number;
+  suggestions: Array<string>;
+  shortDescription: string;
+}
+
+declare interface Analysis {
+  BASEFORM: string;
+  CLASS: string;
+  FSTOUTPUT: string;
+  NUMBER: string;
+  SIJAMUOTO: string;
+  STRUCTURE: string;
+  WORDBASES: string;
+}
+
+declare interface Token {
+  type: TokenType;
+  text: string;
+}
+
+type TokenType = 'WORD' | 'WHITESPACE' | 'PUNCTUATION';
+
+declare interface Sentence {
+  text: string;
+  nextStartType: NextStartType;
+}
+
+type NextStartType = 'PROBABLE' | 'NONE';
```

Comment:
Missing possible types:
```suggestion
type NextStartType = 'PROBABLE' | 'NONE' | 'POSSIBLE' | 'NO_START';
```